test/serializable_resource_test.rb

```diff
@@ -71,6 +71,13 @@ def test_serializable_resource_with_collection_containing_errors
         }
         assert_equal serializable_resource.as_json(options), expected_response_document
       end
+
+      def test_serializable_resource_when_no_serializer_found
+        resource = ModelWithErrors.new
+        serializable_resource = ActiveModel::SerializableResource.new(resource)
+        # TODO define error to raise here
+        assert_raises { serializable_resource.as_json }
```

Comment:
I'm thinking in terms of the principle of least surprise.  Would you be more surprised if we just call to_json etc on the object or if it blows up?

I really don't like the idea of it blowing up, unless the message were: ' did you check serializer? first? Did you mean to pass in the serializer name? Did you forget to make a serializer?'

I think the it may be least surprising if it behaves like the controller, and just didn't wrap the object in a serializer/adapter, or at least a 'pass through' adapter.

i.e. The controller says: can I serialize you ? Let's do this: pass through, as you wre

B mobile phone

> On Mar 11, 2016, at 6:22 AM, L. Preston Sego III <notifications@github.com> wrote:
> 
> In test/serializable_resource_test.rb:
> 
> > @@ -71,6 +71,13 @@ def test_serializable_resource_with_collection_containing_errors
> >          }
> >          assert_equal serializable_resource.as_json(options), expected_response_document
> >        end
> > +
> > +      def test_serializable_resource_when_no_serializer_found
> > +        resource = ModelWithErrors.new
> > +        serializable_resource = ActiveModel::SerializableResource.new(resource)
> > +        # TODO define error to raise here
> > +        assert_raises { serializable_resource.as_json }
> maybe "No serializer found for #{model_name}"
> 
> Though, I wonder, do we have tests for no-serializer? cause it should just use the default json renderer shouldn't it?
> 
> or are we wanting to explicity force the use of serializers? I could see that being useful for JSON API.. but... idk.
> 
> —
> Reply to this email directly or view it on GitHub.
>